### PR TITLE
SB-21886 Allowing locationCodes as array of map format which contains…

### PR DESCRIPTION
**Pull request by Hari-stackroute**
… type and code of location

### CODE REVIEW

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 1]
```diff
@@ -64,7 +64,17 @@ private void validateLocationCodes(Request userRequest) {
           ERROR_CODE);
     }
     if (locationCodes != null) {
-      List<String> set = new ArrayList(new HashSet<>((List<String>) locationCodes));
+      List<String> set = null;
+      if (((List) locationCodes).get(0) instanceof String) {
+        set = new ArrayList(new HashSet<>((List<String>) locationCodes));
```
Comment (reshmi-nair) [inline actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:69]: Request is having List<String>, so is new ArrayList() required? Also ad comments to specify why this instance of check is required

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 2]
```diff
@@ -64,7 +64,17 @@ private void validateLocationCodes(Request userRequest) {
...
+        userRequest.getRequest().put(JsonKey.LOCATION_CODES, set);
+      } else {
+        set = new ArrayList();
+        List<Map<String, String>> locationList = (List<Map<String, String>>) locationCodes;
```
Comment (reshmi-nair) [inline actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:73]: Validation for type and code need to be added

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java (hunk 1 above)
Comment (Hari-stackroute) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:69]: @reshmi-nair I agree with the point, but it is existing code where list is converted to set in-order to remove duplicate location codes if it is sent and again it is converted to list.

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java (hunk 2 above)
Comment (Hari-stackroute) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:73]: @reshmi-nair i thought once we take school into consideration we can check all the types what all we decided.

Comment (Hari-stackroute) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:74]: now we are validating the location type @reshmi-nair

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java (hunk 1 above)
Comment (Hari-stackroute) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:70]: As of now we are allowing locationCodes as array of string and Map of list which contains type and code as keys in the map entry, So in order to distinguish the difference in the input we are checking the first element in the list.

**Review by reshmi-nair — APPROVED** [review]

**Pull request merged**